src/actions/genSale/getSaleStatus.ts

```diff
@@ -0,0 +1,34 @@
+import { BigNumber } from "ethers/lib/ethers";
+import { GenSale } from "../../contracts/types";
+import { GenSaleStatus } from "../../types";
+
+export const getSaleStatus = async (contract: GenSale) => {
+    const saleStarted = await contract.saleStarted();
+
+    if (!saleStarted) {
+        return GenSaleStatus.NotStarted;
+    }
+
+    const saleDataPromises = [
+        contract.domainsSold(),
+        contract.amountForSale(),
+        contract.salePrice(),
+    ];
+
+    const [
+        numSold,
+        totalForSale,
+        price
+    ] = await Promise.all(saleDataPromises)
+
+    if (numSold.gte(totalForSale)) {
+        return GenSaleStatus.Ended;
+    }
+
+    if (price.gt(BigNumber.from(0))) {
```

Comment:
BigNumber functions take a `BigNumberish`, which is something that can be converted to a BigNumber, so you don't have to manually do `BigNumber.from()` for this and other functions

Reply:
removed bignumber from